analytic_hr_department_restriction/security/security_rules.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="group_analytic_accounting_without_department" model="res.groups">
```

Comment:
AFAIK, this is working because Odoo itself is not adding any other record rules, but if this changes in other subsequent modules or in the core, this won't work, being record rules with groups additive. Can you please do this overriding the ORM code like `sales_team_security` and other similar modules?

Reply:
Sorry, but I don't understand what the problem is in this case, i.e. we want to limit the analytical accounts/plans that are displayed. If in the future some other module (or core) adds some restriction (for example with followers) I don't think you should change that behavior, I think you should add this condition as an additive rule and not completely redefine the rules by “omitting” the rules that already exist, but it's just my opinion.

Reply:
But this is not how record rules with groups work. When there are several record rules, the result is the sum of both recordset results, which is not what we want. We always want to show only the department|empty ones, no matter if other modules change this.

Reply:
I still do not understand the problem, i.e. the behavior is as follows:
- Users without “extra” permissions: will only see analytic plans/accounts that have department defined and is the department of the user's employee.
- Users with the “Analytic Accounting (without department)” permission: Only if you have this permission, you will additionally be able to view plans/accounts that have NO department defined.
- Users with the Invoking/Billing permission: Will be able to view all analytic plans/accounts without restriction.